src/Payments/PaymentContext.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Payment context.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\Payments;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * PaymentContext class.
+ */
+class PaymentContext {
+	/**
+	 * Payment method ID.
+	 *
+	 * @var string
+	 */
+	protected $payment_method = '';
+
+	/**
+	 * Order object for the order being paid.
+	 *
+	 * @var \WC_Order
+	 */
+	protected $order;
+
+	/**
+	 * Holds data to send to the payment gateway to support payment.
+	 *
+	 * @var array Key value pairs.
+	 */
+	protected $payment_data = [];
+
+	/**
+	 * Magic getter for protected properties.
+	 *
+	 * @param string $name Property name.
+	 */
+	public function __get( $name ) {
+		if ( in_array( $name, [ 'payment_method', 'order', 'payment_data' ], true ) ) {
+			return $this->$name;
+		}
+		return null;
```

Comment:
What about throwing an error if something tries to access a property that isn't exposed?

Reply:
PHP doesn't throw exceptions if you access non-defined variables so I don't think we need to here. This is just for convenience -- I was going to just use stdClass but wanted a little extra flexibility just in case.